File: WindowsFormsApplicationLab1/WindowsFormsApplicationLaba1/Stove.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApplicationLaba1
{
    class Stove
    {
        private Pan pan;

        private bool state = false;
        public bool State { set { state = value; } get { return state; } }

        public Pan Pan { set { pan = value; } get { return pan; } }

        public void Cook()
        {
                while (!pan.IsReadyPan())
                {
                    pan.Cook();
                }       
        }
    }
}

File: TPlaba1/src/Stove.java

public class Stove {
	private Pan pan;

	public Pan GetPan() {
		return pan;
	}

	public void SetPan(Pan p) {
		pan = p;
	}

	private boolean state = false;

	public boolean getState() {
		return state;
	}

	public void setState(boolean s) {
		state = s;
	}

	public void Cook() {
		if (state) {
			while (!pan.IsReadyPan()) {
				pan.Cook();
			}
		}
	}
}
